lib/auth/trustedcluster_test.go

```diff
@@ -391,6 +393,30 @@ func TestValidateTrustedCluster(t *testing.T) {
 			[]types.CertAuthType{resp.CAs[0].GetType(), resp.CAs[1].GetType(), resp.CAs[2].GetType()},
 		)
 	})
+
+	t.Run("trusted clusters prevented on cloud", func(t *testing.T) {
+		beforeModules := modules.GetModules()
+		modules.SetModules(&modules.TestModules{
+			TestFeatures: modules.Features{Cloud: true},
+		})
+		t.Cleanup(func() { modules.SetModules(beforeModules) })
+
+		req := &ValidateTrustedClusterRequest{
+			Token: "invalidtoken",
+			CAs:   []types.CertAuthority{},
+		}
+
+		server := ServerWithRoles{authServer: a}
+		_, err := server.ValidateTrustedCluster(ctx, req)
+		require.ErrorIs(t, err, trace.NotImplemented("cloud clusters do not support trusted cluster resources"))
```

Comment:
```suggestion
		assert.True(t, trace.IsNotImplemented(err), "ValidateTrustedCluster returned an unexpected error, got = %v (%T), want trace.NotImplementedError", err, err)
```

So we don't depend on the exact error message?

Also: https://github.com/golang/go/wiki/TestComments#keep-going.